### PR TITLE
Prepare for the Path resolver change to GET

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Prepares for the Cloud API change, together with a fallback for the old API to avoid problems during migration.
- This PR should be merged before the https://github.com/enso-org/cloud-v2/pull/1236 PR is _deployed_.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: tools/http-test-helper/src/main/java/org/enso/shttp/cloud_mock/PathResolver.java [hunk 1]
```diff
@@ -27,8 +26,19 @@ public boolean canHandle(String subPath) {
...
+      return;
+    }
+
+    String prefix = "path=";
+    if (!queryString.startsWith(prefix)) {
+      exchange.sendResponse(400, "Missing `path` parameter in query string: `" + queryString + "`");
+      return;
+    }
+
+    String path = queryString.substring(prefix.length());
```
Comment (indiv0) [inline tools/http-test-helper/src/main/java/org/enso/shttp/cloud_mock/PathResolver.java:41]: Don't you still need to URI-decode the path at this point?

Comment (radeusgd) [reply tools/http-test-helper/src/main/java/org/enso/shttp/cloud_mock/PathResolver.java:41]: Java's `URI::getQuery` returns an already decoded query fragment. `getRawQuery` is undecoded.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Existing_Enso_Asset.enso [hunk 2]
```diff
@@ -82,12 +84,17 @@ type Existing_Enso_Asset
...
+        r = Utils.http_request_as_json HTTP_Method.Get uri error_handlers=error_handlers
+        ## Workaround to keep compatibility with old cloud API
+           TODO remove it after https://github.com/enso-org/cloud-v2/pull/1236 has been deployed
+        response = r.catch Enso_Cloud_Error error-> case error of
+            Enso_Cloud_Error.Unexpected_Service_Error status _ ->
+                if status.code != 404 then r else
+                    old_uri = (URI.from Utils.cloud_root_uri) / "path/resolve"
+                    old_payload = JS_Object.from_pairs [["path", path]]
+                    Context.Output.with_enabled <| Utils.http_request_as_json HTTP_Method.Post old_uri old_payload error_handlers=error_handlers
+            _ -> r
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Existing_Enso_Asset.enso:97]: Temporary fallback to make the migration smooth: if the GET endpoint responds with 404, we fall back to the POST endpoint.

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Existing_Enso_Asset.enso:97]: The commit d8460d0d063216ec2d53101c7d865da3199f2173 should be reverted after https://github.com/enso-org/cloud-v2/pull/1236 is deployed.

**Review by AdRiley — APPROVED** [review]

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**